py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -121,6 +124,52 @@ def setup_engine(self) -> None:
         self.engine = runtime.deserialize_cuda_engine(self.serialized_engine)
         self.context = self.engine.create_execution_context()
 
+        if self.settings.strip_engine_weights:
```

Comment:
We likely shouldnt be doing the refit **in** these modules

I think for weight stripping there are 2 workflows.

1. a user just wants a weight stripped engine. They should use `convert_exported_program_to_trt_engine` with settings `strip_weights`. The choice of `make_refittable` can be used to decide between `kREFIT` and `kREFIT_IDENTICAL` (though it might not be entirely clear so we might want to think about that setting). 
2. We want to utilize weight stripping to have a lighter weight cache. Here this choice is opaque to the user. The user choice of `make_refittable` controls if we use `kREFIT` or `kREFIT_IDENTICAL`. But once the engine is loaded or we pull from cache we immediately refit (prior to passing the engine to the TRTModule).  
3. The user wants a stripped weights compiled program (im not sure why or if this is a real usecase). Here, this is basically the same as lazy engine loading. We would require that users need to run through `refit_engine_weights` before executing.

Reply:
Got it. The very beginning idea/design is commented below. I'll move the refitting part back to TRTInterpreter.run()

> The choice of `make_refittable` can be used to decide between `kREFIT` and `kREFIT_IDENTICAL`

Do you mean we use `make_refittable` to control both `kREFIT` and `kREFIT_IDENTICAL`?